lib/rules/no-useless-rename.js

```diff
@@ -97,27 +105,19 @@ module.exports = {
 
             for (const property of node.properties) {
 
-                /*
-                 * TODO: Remove after babel-eslint removes ExperimentalRestProperty
-                 * https://github.com/eslint/eslint/issues/12335
-                 */
-                if (property.type === "ExperimentalRestProperty") {
-                    continue;
-                }
-
                 /**
                  * Properties using shorthand syntax and rest elements can not be renamed.
                  * If the property is computed, we have no idea if a rename is useless or not.
                  */
-                if (property.shorthand || property.type === "RestElement" || property.computed) {
+                if (property.type !== "Property" || property.shorthand || property.computed) {
```

Comment:
I made this change to be consistent with the condition in `fix()` (where it expects only `Property` or module specifiers), so checking for `ExperimentalRestProperty`  above became unnecessary.